### PR TITLE
nixos/starship: change module logic that sets STARSHIP_CONFIG to not override user config

### PR DESCRIPTION
**Pull request by reedriley**
The config for the Starship module if you set a system-wide config sets a `STARSHIP_CONFIG` environment variable; that will then override any user-level configs.  (e.g., any created using `home-manager`.)

I can't imagine a scenario where we'd want to ignore a user-provided config, so this patch changes the module to skip setting this variable if there's a user-specific config in the standard location.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/starship.nix
Comment (teto) [inline nixos/modules/programs/starship.nix:47]: this looks weird, aint that an XDG path ? shouldn't it be using `${XDG_CONFIG_HOME:-.config}` ? is it possible to ask the starship executable instead what settings it loaded ?

Does  the home-manager module export STARSHIP_CONFIG ? in which case nixos could just set it conditionnally.

Comment (reedriley) [reply nixos/modules/programs/starship.nix:47]: You're probably right about the `XDG_CONFIG_HOME` - but the `starship` config documentation has this listed as a static path here: https://starship.rs/config/

I just went and double checked [the source code](https://github.com/starship/starship/blob/686bda1706e5b409129e6694639477a0f8a3f01b/src/configure.rs#L651) as well and I think it might actually be hard-coded to the home directory?

Home manager doesn't export `STARSHIP_CONFIG` in it's module; it just creates the config file at the right path.

How about a compromise?  I'll go and amend this to include a comment referencing the fact that starship appears to use a hardcoded config path - so our descendants will understand why this is a little weird?

Comment (teto) [reply nixos/modules/programs/starship.nix:51]: ty for checking, sounds about right. A comment should do the trick but please ping the starship maintainer I dont feel confident to merge something like this as an outsider.

**Comment by reedriley** [comment]
I'm happy to tweak this or discuss further - what needs to happen to unblock this?

**Comment by teto** [comment]
find the maintainer and if it's not mentioned in the module, use git blame to see who usually edits the file.

**Comment by drupol** [comment]
What did I do ?!

**Comment by reedriley** [comment]
> What did I do ?!

Sorry!  Absolutely nothing as far as I know?  I'm just tagging folks who might be able to help review this?

**Comment by drupol** [comment]
Oh, OK! :)
I'll leave this review to the maintainer, since it involves more specific things.

**Review by marsam — APPROVED** [review]

**Pull request merged**
